intel_extension_for_transformers/llm/quantization/nn/modules.py

```diff
@@ -134,18 +145,100 @@ def forward(self, x: torch.Tensor):
+            self.compute_dtype,
+            self.weight_dtype,
+            self.scale_dtype,
+            False,
+        )
         weight.resize_(shape)
-        self.weight = ParamsQBits(data=weight,
-                                  requires_grad=False,
-                                  quant_state={"scheme": self.scheme},
-                                  blocksize=self.blocksize,
-                                  compress_statistics=self.compress_statistics,
-                                  quant_dtype=self.weight_dtype,
-                                  scale_dtype=self.scale_dtype)
+        self.weight = ParamsQBits(
+            data=weight,
+            requires_grad=False,
+            quant_state={"scheme": self.scheme},
+            blocksize=self.blocksize,
+            compress_statistics=self.compress_statistics,
+            quant_dtype=self.weight_dtype,
+            scale_dtype=self.scale_dtype,
+        )
         if bias is not None:
             self.bias = torch.nn.Parameter(bias, requires_grad=False)
 
+    def set_gptq_weights_bias(
+        self,
+        int_weight,
+        gptq_scales,
+        gptq_zeros,
+        g_idx,
+        q_config,
+        n_head=0,
+        n_head_kv=0,
+        permute_func=None,
+        bias=None,
+    ):
+        if permute_func:
+            int_weight = (
+                permute_func(int_weight.t(), n_head, n_head_kv).t().contiguous()
+            )
+            gptq_scales = (
+                permute_func(gptq_scales.t(), n_head, n_head_kv).t().contiguous()
+            )
+            gptq_zeros = (
+                permute_func(gptq_zeros.t(), n_head, n_head_kv).t().contiguous()
+            )
+
+        if q_config.gptq_quantize_config["desc_act"]:
+            int_weight2 = int_weight.clone()
+            group_size = q_config.gptq_quantize_config["group_size"]
+            group_dict = {}
+            for i in range(len(g_idx)):
+                group_idx = g_idx[i].item()
+                if group_idx not in group_dict:
+                    target_idx = group_idx * group_size
+                    group_dict[group_idx] = 0
+                else:
+                    group_dict[group_idx] = group_dict[group_idx] + 1
+                    target_idx = group_idx * group_size + group_dict[group_idx]
+                int_weight2[target_idx] = int_weight[i]
+            int_weight = int_weight2
+
+        if q_config.gptq_quantize_config["bits"] == 4:
+            int_weight = (int_weight - 8) * 16
+            gptq_scales = gptq_scales / 16
+            gptq_zeros = (gptq_zeros - 8) * 16
+        if q_config.gptq_quantize_config["sym"]:
+            gptq_zeros = torch.empty(0, dtype=torch.int8)
+
+        if not q_config.gptq_quantize_config["desc_act"]:
+            g_idx = torch.empty(0, dtype=torch.int32)
+
+        packw = torch.ops.bestlaop.woq_packq(
+            int_weight.contiguous(),
+            gptq_scales.float().contiguous(),
+            gptq_zeros,
```

Comment:
emm, i guess all tensors passed to `bestlaop.packq` should be contiguous, incuding zp & g_idx

Reply:
fix :)